### PR TITLE
icx: new module icx_user

### PR DESCRIPTION
**Pull request by sushma-alethea**
SUMMARY
Running command to manage ICX device by Ansible automatically.

ISSUE TYPE
New Module Pull Request

COMPONENT NAME
ICX_7150

ADDITIONAL INFORMATION
Achieving of running command to manage ICX device by Ansible automatically.

### CODE REVIEW

File: lib/ansible/modules/network/icx/icx_user.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+        exclusive with the C(name) argument.
+    aliases: ['users', 'collection']
+    type: list
+    suboptions:
+      name:
+        description:
+          - The username to be configured on the ICX device.
+            This argument accepts a string value and is mutually exclusive
+            with the C(aggregate) argument.
+            Please note that this option is not same as C(provider username).
```
Comment (NilashishC) [inline lib/ansible/modules/network/icx/icx_user.py]: This module does not use the `provider` dict. This line seems to be redundant.

File: lib/ansible/modules/network/icx/icx_user.py [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+        description:
+          - Check running configuration. This can be set as environment variable.
+           Module will use environment variable value(default:True), unless it is overriden, by specifying it as module parameter.
+        type: bool
+  name:
+    description:
+      - The username to be configured on the ICX device.
+        This argument accepts a string value and is mutually exclusive
+        with the C(aggregate) argument.
+        Please note that this option is not same as C(provider username).
```
Comment (NilashishC) [inline lib/ansible/modules/network/icx/icx_user.py]: Same as above.

File: lib/ansible/modules/network/icx/icx_user.py [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+
+import re
+import base64
+import hashlib
+
+from functools import partial
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils.network.common.utils import remove_default_spec
+from ansible.module_utils.connection import Connection, ConnectionError, exec_command
```
Comment (NilashishC) [inline lib/ansible/modules/network/icx/icx_user.py]: `Connection` and `ConnectionError` imports seems to be unused.

File: lib/ansible/modules/network/icx/icx_user.py [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+    argument_spec.update(element_spec)
+
+    mutually_exclusive = [('name', 'aggregate')]
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           mutually_exclusive=mutually_exclusive,
+                           supports_check_mode=True)
+
+    result = {'changed': False}
+    exec_command(module, 'skip')
```
Comment (NilashishC) [inline lib/ansible/modules/network/icx/icx_user.py:369]: Curious to know why do we need this call here?

File: lib/ansible/modules/network/icx/icx_user.py [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: icx_user
+version_added: "2.9"
+author: "Ruckus Wireless (@Commscope)"
+short_description: Manage the user accounts on Ruckus ICX 7000 series switches.
+description:
```
Comment (ganeshrn) [inline lib/ansible/modules/network/icx/icx_user.py:19]: Add a notes section for icx version tested against and link to platform guide
In the notes section, you can point to the icx platform guide https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/network/user_guide/platform_icx.rst

Refer: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_interfaces.py#L107
Applicable for other icx modules as well.

File: lib/ansible/modules/network/icx/icx_user.py (hunk 4 above)
Comment (sushma-alethea) [reply lib/ansible/modules/network/icx/icx_user.py:369]: we need to run "skip" to disable page display mode.

**Review by ganeshrn — APPROVED** [review]

**Review by NilashishC — APPROVED** [review]

**Review by ganeshrn — APPROVED** [review]

**Pull request merged**